EPMLSTRCMW-281 Rename ProjectFileConfiguration to WdlParams

Nikolai-Isaev opened the pull request:
refactor: Rename ProjectFileConfiguration to WdlParams

ProjectConfiguration constructor now accepts single ProjectFileConfiguration instance.
ProjectFileConfiguration class with all corresponding references was renamed to WdlParams

Nikolai-Isaev committed:
refactor: Rename ProjFileConfig to WdlParams

- ProjectConfiguration constructor now accepts single
ProjectFileConfiguration instance.
- ProjectFileConfiguration class with all corresponding references
was renamed to WdlParams

DruzhininPavel commented on services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala:
```diff
@@ -31,21 +25,21 @@ object ProjectConfigurationService {
 
-        def toProjectFileConfiguration(path: Path, inputs: Map[String, TypedValue]): ProjectFileConfiguration =
-          ProjectFileConfiguration(path, inputs.map((FileParameter.apply _).tupled).toList)
+        def toProjectFileConfiguration(path: Path, inputs: Map[String, TypedValue]): WdlParams =
```
I think do you mean ?
```suggestion
        def toWdlParams(path: Path, inputs: Map[String, TypedValue]): WdlParams =
```

Nikolai-Isaev replied:
Oops, fixed

DruzhininPavel reviewed (changes requested):
Looks good, but some changes need.

gominasian reviewed (approved):
Nice!

DruzhininPavel reviewed (approved):
Well done.

myazinn commented on repositories/src/test/scala/cromwell/pipeline/datastorage/dao/utils/GeneratorUtils.scala:
```diff
@@ -124,18 +124,18 @@ object GeneratorUtils {
     projectId <- projectIdGen
     active <- booleanGen
-    projectFileConfigurations <- listOfN(projectFileConfigurationGen)
+    projectFileConfiguration <- wdlParamsGen
```
```suggestion
    wdlParams <- wdlParamsGen
```

Nikolai-Isaev replied:
Double shift search doesn't like me at all (
Fixed, postman too

myazinn commented on repositories/src/main/scala/cromwell/pipeline/datastorage/dto/CromwellInput.scala:
```diff
@@ -7,5 +7,5 @@ final case class CromwellInput(
   projectVersion: PipelineVersion,
   files: List[ProjectFile],
-  configurations: List[ProjectFileConfiguration]
+  configuration: WdlParams
```
I think we should rename it to `wdlParams` for consistency as well

Nikolai-Isaev replied:
Fixed

myazinn commented on services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala:
```diff
@@ -31,21 +25,21 @@ object ProjectConfigurationService {
           case Some(config) =>
-            val oldFileConfigsMap = config.projectFileConfigurations.map(toTuple).toMap
-            val newFileConfigsMap = projectConfiguration.projectFileConfigurations.map(toTuple).toMap
+            val oldWdlParamsMap = List(config.wdlParams).map(toTuple).toMap
+            val newWdlParamsMap = List(projectConfiguration.wdlParams).map(toTuple).toMap
```
This definitely can be simplified. Why do you need to convert single configuration to List and then to tuple and then to Map if you know that you have only one configuration and its path is known?

Nikolai-Isaev replied:
I believe that that whole logic may be redundant here from now on. And just
```
def addConfiguration(projectConfiguration: ProjectConfiguration, userId: UserId): Future[Unit] =
        repository.addConfiguration(projectConfiguration)
```
will do just fine. Am I right?

Nikolai-Isaev replied:
Tried to refactor this a little

myazinn commented on services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala:
```diff
@@ -31,21 +25,21 @@ object ProjectConfigurationService {
-            val updatedFileConfigsMap = oldFileConfigsMap ++ newFileConfigsMap
-            val updatedFileConfigs = updatedFileConfigsMap.map((toProjectFileConfiguration _).tupled).toList
+            val updatedWdlParamsMap = oldWdlParamsMap ++ newWdlParamsMap
+            val updatedWdlParams = updatedWdlParamsMap.map((toWdlParams _).tupled).head
```
Please avoid using unsafe operations like `head` or `get` in your code. Usually it shows that you are doing something wrong

Nikolai-Isaev replied:
Removed this piece

myazinn reviewed (changes requested):
What about postman collection?

myazinn commented on postman/Configurations.postman_collection.json:
```diff
@@ -18,7 +18,7 @@
 				"body": {
 					"mode": "raw",
-					"raw": "{\n    \"id\": \"\",  \r\n    \"projectFileConfigurations\": [\r\n        {\"path\": \"\", \"inputs\": [\r\n            {\"name\": \"\", \"typedValue\": {\"_type\": \"\", \"value\": \"\"}},\r\n            {\"name\": \"\", \"typedValue\": {\"_type\": \"\", \"value\": \"\"}}\r\n            ]}\r\n    ]\r\n}",
+					"raw": "{\n    \"id\": \"\",  \r\n    \"wdlParams\": [\r\n        {\"path\": \"\", \"inputs\": [\r\n            {\"name\": \"\", \"typedValue\": {\"_type\": \"\", \"value\": \"\"}},\r\n            {\"name\": \"\", \"typedValue\": {\"_type\": \"\", \"value\": \"\"}}\r\n            ]}\r\n    ]\r\n}",
```
I don't think that you've made enough changes. Try to upload configuration using this example and see what happens

Nikolai-Isaev replied:
Fixed: moved `Build project configuration by project` from files to configurations in postman collections. Path edited

myazinn replied:
Nice! But I wasn't talking about moving another endpoint to this collection (although it is definetely needed too). This particular line still contains mistake.

Nikolai-Isaev replied:
"Add configuration" request body (raw) is now updated to correspond to changed `ProjectConfiguration`structure (only one `wdlParams` instead of seq). Also changed a little to look pretty in postman.
Hope I got it at last :)

myazinn commented on services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala:
```diff
@@ -22,72 +22,43 @@ class ProjectConfigurationServiceTest extends AsyncWordSpec with Matchers with M
         configurationService.addConfiguration(activeConfiguration, userId).failed.map(_ shouldBe error)
       }
-
-      "return failure if update wasn't successful" in {
```
I think we should keep this test case

Nikolai-Isaev replied:
It doesn't depend on the update method anymore. Which condition should be covered by this testcase?

myazinn replied:
You are right, I haven't looked carefully. Closing this conversation.

Nikolai-Isaev commented:
"Add configuration" request body (raw) is now updated to correspond to changed ProjectConfigurationstructure (only one wdlParams instead of seq). Also changed indentation a little to look pretty in postman.
Hope I got it at last :)
